### ISSUE TITLE
Target an image handle for auto generation only inside a matrix

### ISSUE THREAD

**Issue by alexroper** [spacecatninja/craft-imager-x#39]
### I'm submitting a... 

- [ ] bug report
- [ ] feature request
- [x] general question

### Description

Is it possible to target an image handle inside a specific matrix field for auto transform generation? Can I do this without targeting all fields with the same handles?

Say I have a matrix field named `content` with a block type named `carousel` and a field inside that with the handle of `image`.

This seems to work:

```
'fields' => [
    'image' => ['myTransform'],
],
```

But this doesn't:

```
'fields' => [
    'content:carousel.image' => ['myTransform'],
],
```

The format that uses `<matrixFieldHandle>:<matrixTypeHandle>.<fieldHandle>` only seems to work with elements.

```
'elements' => [
    [
        'elementType' =>  \craft\elements\Entry::class,
        'criteria' => [
            'section' => ['news'],
        ],
        'fields' => ['content:carousel.image'],
        'transforms' => ['myTransform']
    ],
],
```
Is this working correctly or am I doing something wrong?

### Additional info

- Imager version & edition: Pro v3.0.9
- Imager transformer: imagemagick
- Craft version: 3.4.18
- PHP version: 7.3.8
- Image driver & version: Imagick 3.4.3 (ImageMagick 6.9.6-2)
- Database driver & version: MySQL 5.7.26
- Plugins & versions (if relevant):

**Comment by aelvan** [comment]
No, you're right, the matrix field handle syntax only works inside the elements functionality. Supporting it in the general fields functionality was a bit tricky, so I dropped it initially. But I'll leave this open and revisit it down the line.

FWIW, if you drop the criteria in an elements definition, it'll do the same thing. But you'll have to define it for each element type of course.

**Comment by alexroper** [comment]
Thanks. I thought that's how it worked but just wanted to make sure I wasn't missing something.

I realized this wouldn't be too painful if I just combine several sections in the element criteria. Didn't occur to me I could do that at first.

```php
return [
    'elements' => [
        [
            'elementType' => \craft\elements\Entry::class,
            'criteria' => [
                'section' => ['news', 'about', 'blog', 'contact'],
            ],
            'fields' => ['pageContent:image.image'],
            'transforms' => ['imageBlock']
        ],
    ]
];
```

**Comment by daniellelecomte** [comment]
Is it possible to target an image inside of a super-table field in the same way that you can for a matrix field?
`<matrixFieldHandle>:<matrixTypeHandle>.<fieldHandle>`

**Comment by alexroper** [comment]
@daniellelecomte I don't think so. My experiments with targeting image transforms inside a Super Table field haven't worked at all. But maybe I'm doing something wrong. ¯\_(ツ)_/¯

**Comment by aelvan** [comment]
@daniellelecomte It's not at the moment, but shouldn't bbe too difficult to add. File a separate FR if you want me to look into it.

**Comment by aelvan** [comment]
Amazingly enough, when adding support for content block fields, I also found a solution for this. Better late than never, I guess. 😭

All of these should work now:
```
'fields' => [
    'image' => ['myTransform'],
    'contentBlock.image' => ['myTransform'],
    'myMatrix:entryType.image' => ['myTransform'],
],
```
